### PR TITLE
[Keyboard] Add Kangaroo 

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Add Kangaroo keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xelus/kangaroo/kangaroo.c [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include "kangaroo.h"
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/kangaroo.c:1]: Could you add a license header here, please?

Here's a sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/xelus/kangaroo/kangaroo.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/kangaroo.h:1]: Could you add a license header here, please?

File: keyboards/xelus/kangaroo/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/keymaps/default/keymap.c:1]: Could you add a license header here, please?

File: keyboards/xelus/kangaroo/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/keymaps/via/keymap.c:1]: Could you add a license header here, please?

File: keyboards/xelus/kangaroo/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+#   comment out to disable the options.
+#
+
+BOOTMAGIC_ENABLE = lite	# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes	# Mouse keys
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = yes 	# Console for debug
+COMMAND_ENABLE = yes    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	    # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/rules.mk:14]: Switching out the tabs for spaces here.

```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes           # USB Nkey Rollover
```

File: keyboards/xelus/kangaroo/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+	  [0] = LAYOUT_all(
+	    KC_MUTE, KC_VOLD, KC_VOLU, KC_TRNS,   KC_PSCR, KC_SLCK, KC_PAUS,   KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+	    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,   KC_INS,  KC_HOME, KC_PGUP,   KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,
+	    KC_PPLS, KC_P7,   KC_P8,   KC_P9,     KC_DEL,  KC_END,  KC_PGDN,   KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+	    KC_PPLS, KC_P4,   KC_P5,   KC_P6,                                  KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+	    KC_PENT, KC_P1,   KC_P2,   KC_P3,              KC_UP,              KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+	    KC_PENT, KC_PDOT, KC_P0,   KC_P0,     KC_LEFT, KC_DOWN, KC_RGHT,   KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RGUI, KC_MENU, KC_RCTL),
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/keymaps/default/keymap.c:10]: The Windows Context Menu key is actually `KC_APP`.

```suggestion
	    KC_MUTE, KC_VOLD, KC_VOLU, KC_TRNS,   KC_PSCR, KC_SLCK, KC_PAUS,   KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
	    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,   KC_INS,  KC_HOME, KC_PGUP,   KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,
	    KC_PPLS, KC_P7,   KC_P8,   KC_P9,     KC_DEL,  KC_END,  KC_PGDN,   KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
	    KC_PPLS, KC_P4,   KC_P5,   KC_P6,                                  KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
	    KC_PENT, KC_P1,   KC_P2,   KC_P3,              KC_UP,              KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
	    KC_PENT, KC_PDOT, KC_P0,   KC_P0,     KC_LEFT, KC_DOWN, KC_RGHT,   KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RGUI, KC_APP,  KC_RCTL),
```

File: keyboards/xelus/kangaroo/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	  [0] = LAYOUT_all(
+	    KC_MUTE, KC_VOLD, KC_VOLU, KC_TRNS,   KC_PSCR, KC_SLCK, KC_PAUS,   KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
+	    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,   KC_INS,  KC_HOME, KC_PGUP,   KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,
+	    KC_PPLS, KC_P7,   KC_P8,   KC_P9,     KC_DEL,  KC_END,  KC_PGDN,   KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+	    KC_PPLS, KC_P4,   KC_P5,   KC_P6,                                  KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
+	    KC_PENT, KC_P1,   KC_P2,   KC_P3,              KC_UP,              KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
+	    KC_PENT, KC_PDOT, KC_P0,   KC_P0,     KC_LEFT, KC_DOWN, KC_RGHT,   KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RGUI, KC_MENU, KC_RCTL),
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/keymaps/via/keymap.c:10]:
```suggestion
	    KC_MUTE, KC_VOLD, KC_VOLU, KC_TRNS,   KC_PSCR, KC_SLCK, KC_PAUS,   KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
	    KC_NLCK, KC_PSLS, KC_PAST, KC_PMNS,   KC_INS,  KC_HOME, KC_PGUP,   KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_BSPC,
	    KC_PPLS, KC_P7,   KC_P8,   KC_P9,     KC_DEL,  KC_END,  KC_PGDN,   KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
	    KC_PPLS, KC_P4,   KC_P5,   KC_P6,                                  KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,
	    KC_PENT, KC_P1,   KC_P2,   KC_P3,              KC_UP,              KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
	    KC_PENT, KC_PDOT, KC_P0,   KC_P0,     KC_LEFT, KC_DOWN, KC_RGHT,   KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, KC_RGUI, KC_APP,  KC_RCTL),
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also needs an `info.json` file, but I can do that if you're not up for it.

File: keyboards/xelus/kangaroo/rules.mk [hunk 8]
```diff
@@ -5,16 +5,16 @@ MCU = STM32F072
...
-MOUSEKEY_ENABLE = yes	# Mouse keys
-EXTRAKEY_ENABLE = yes	# Audio control and System control
-CONSOLE_ENABLE = yes 	# Console for debug
-COMMAND_ENABLE = yes    # Commands for debug and configuration
-SLEEP_LED_ENABLE = no   # Breathing sleep LED during USB suspend
-NKRO_ENABLE = yes	    # USB Nkey Rollover
-RGBLIGHT_ENABLE = no    # Enable keyboard RGB underglow
-BLUETOOTH_ENABLE = no   # Enable Bluetooth
-AUDIO_ENABLE = no       # Audio output
+BOOTMAGIC_ENABLE = lite	    # Virtual DIP switch configuration
```
Comment (noroadsleft) [inline keyboards/xelus/kangaroo/rules.mk:8]: I guessed I missed a tab on the previous suggestion.

```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**